### PR TITLE
Fix application /:sid return

### PR DESCRIPTION
**Pull request by khanghugo**
Tested on the current webapp and it works fine.

I was looking around to see if the current webapp can pull this correctly or not but the change seems to bring some more changes.

### CODE REVIEW

**Comment by khanghugo** [comment]
On that note, it works well with the `refresh` branch of the webapp.

File: lib/routes/api/applications.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ router.get('/:sid', async(req, res) => {
     const account_sid = req.user.hasAccountAuth ? req.user.account_sid : null;
     const results = await Application.retrieve(req.params.sid, service_provider_sid, account_sid);
     if (results.length === 0) return res.status(404).end();
-    return res.status(200).json(results);
+    return res.status(200).json(results[0]);
```
Comment (kitajchuk) [inline lib/routes/api/applications.js:107]: In all fairness I feel like the SQL query should be returning a single entry so this feels like maybe also a hack?

Comment (khanghugo) [reply lib/routes/api/applications.js:107]: Funny you mention that, all of `retrieve()` methods every routes happen to do this. Only one, until now, doesn't have and that is `Application`

**Comment by khanghugo** [comment]
This one change breaks 8 tests. I did try revert it and it passes all.

File: lib/routes/api/applications.js (hunk 1 above)
Comment (kitajchuk) [reply lib/routes/api/applications.js:107]: Ya I believe I'm thinking about this wrong lol, at least at the SQL level. So actually this makes sense lol. I suppose the queries will always technically return rows and its up to the application code to return all of them or just one of them like you have here. Sorry, my brain was not quite ready for SQL at the moment lol.

**Review by kitajchuk — APPROVED** [review]

**Comment by khanghugo** [comment]
Let me see if I can fix these tests

File: lib/routes/api/applications.js (hunk 1 above)
Comment (khanghugo) [reply lib/routes/api/applications.js:107]: Haha. I have no idea how SQL works.

Comment (khanghugo) [reply lib/routes/api/applications.js:107]: All I know is select table and see what is going on.

**Comment by kitajchuk** [comment]
> Let me see if I can fix these tests

You totally can. Looking at the tests they all are expecting to get an array so they do the `result[0]` reference. Easy peasy :)

File: lib/routes/api/applications.js (hunk 1 above)
Comment (kitajchuk) [reply lib/routes/api/applications.js:107]: Feel free to push the changes on the webapp refresh branch and I'll take a look at things. I have this api server branch checked out locally so I can test properly 👍

**Pull request merged**